### PR TITLE
Improve the warning message regarding local function not support by pickle

### LINKED ISSUE #947: Local function is not supported by pickle
### 🐛 Describe the bug

I used a local function in a pipeline which results in the warning:

> UserWarning: Local function is not supported by pickle, please use regular python function or functools.partial instead.

So I wrapped the local function in functools.partial and the warning persisted, so I investigated closer. The problem appears to be that `_check_unpickable_fn` is undoing the partial before checking if fn is local, i.e.

This code unwraps partial, converting back to the underlying local function

https://github.com/pytorch/pytorch/blob/301644d3cb8ce7ecb477b5bcb95a7a8a1304fa71/torch/utils/data/datapipes/utils/common.py#L132

And this code then produces the warning

https://github.com/pytorch/pytorch/blob/301644d3cb8ce7ecb477b5bcb95a7a8a1304fa71/torch/utils/data/datapipes/utils/common.py#L136

My calling code is
```

def run(args)

    def batch_transform(x):
        return {"inputs": text_transform(x["text"]), "target": label_transform(x["label"])}

    train_datapipe,, test_datapipe= DATASETS[args.dataset](root=args.data_dir, split=('train', 'test'))
    
    # shuffle and batch
    train_datapipe = train_datapipe.shuffle().batch(args.batch_size).rows2columnar(["text", "label"])
    train_datapipe = train_datapipe.map(partial(batch_transform))

```

This produces the warning regardless of whether batch_transform is partial or not (a fix is to import dill or lift `batch_transform` out of  `run`)

### Versions

PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-58-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 2080 Ti
Nvidia driver version: 525.60.13
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] pytorch-ignite==0.4.10
[pip3] torch==1.13.0+cu117
[pip3] torchdata==0.5.0
[pip3] torchtext==0.14.0
[conda] Could not collect

### PR DESCRIPTION
**Pull request by DongyuXu77**
fixes: https://github.com/pytorch/data/issues/947

### CODE REVIEW

File: torch/utils/data/datapipes/utils/common.py [hunk 1]
```diff
@@ -136,15 +136,15 @@ def _check_unpickable_fn(fn: Callable):
     if _is_local_fn(fn) and not DILL_AVAILABLE:
         warnings.warn(
             "Local function is not supported by pickle, please use "
-            "regular python function or functools.partial instead."
+            "regular python function or ensure dill is available."
```
Comment (ejguan) [inline torch/utils/data/datapipes/utils/common.py:139]:
```suggestion
            "regular python function or ensure dill is available. "
            "If local variables are needed as arguments for the regular function, "
            "please use `functools.partial` to supply them."
```

**Review by ejguan — COMMENTED** [review]
Thank you for this effort. You need to sign EasyCLA before I merge it.

File: torch/utils/data/datapipes/utils/common.py (hunk 1 above)
Comment (DongyuXu77) [reply torch/utils/data/datapipes/utils/common.py]: Thanks for your suggestion

**Comment by DongyuXu77** [comment]
> Thank you for this effort. You need to sign EasyCLA before I merge it.

I have signed several times but still says missing authorization.

**Comment by ejguan** [comment]
@DongyuXu77 It might be the case that your commit is not associated with your email address. Please take a look at https://docs.linuxfoundation.org/v2/easycla/getting-started/easycla-troubleshooting#github-pull-request-is-not-passing

**Comment by ejguan** [comment]
@DongyuXu77 I just checked your commits that are associated with `xudongyu@bupt.edu.com`. Did you sign CLA with this email?

**Comment by DongyuXu77** [comment]
> @DongyuXu77 I just checked your commits that are associated with `xudongyu@bupt.edu.com`. Did you sign CLA with this email?

I found that I make a stupid mistake the correct email is `xudongyu@bupt.edu.cn` instead of `XXX.com`. I tried to change the committed email address, but seems it doesn't work. What should I do to solve that?

**Comment by ejguan** [comment]
> I tried to change the committed email address, but seems it doesn't work. What should I do to solve that?

First thing is to change your config for github.
```bash
git config --global user.name "Name"
git config --global user.email "email"
```
Since you have two commits in the history, you need to do an interactive rebase of the last two commits (choose edit) and amend each commit by
```bash
git commit --amend --author="Name <email>"
git rebase --continue
```

**Pull request closed without merging**